pkg/blobstore/grpcservers/byte_stream_server.go

```diff
@@ -132,6 +163,95 @@ func (s *byteStreamServer) Write(stream bytestream.ByteStream_WriteServer) error
 	})
 }
 
+type zstdReadCloser struct {
+	decoder          *zstd.Decoder
+	underlyingReader io.ReadCloser
+}
+
+func newZstdReaderCloser(underlyingReader io.ReadCloser) (io.ReadCloser, error) {
+	decoder, err := zstd.NewReader(underlyingReader)
+	if err != nil {
+		return nil, err
+	}
+	return &zstdReadCloser{decoder: decoder, underlyingReader: underlyingReader}, nil
+}
+
+func (z *zstdReadCloser) Read(p []byte) (n int, err error) {
+	return z.decoder.Read(p)
+}
+
+func (z *zstdReadCloser) Close() error {
+	z.decoder.Close()
+	return z.underlyingReader.Close()
+}
```

Comment:
This type is quite similar to `zstdReader` that lives in `reference_expanding_blob_access.go`. Do we feel like unifying this, or do we think ["a little copying is better than a little dependency"](https://go-proverbs.github.io)? I'm fine either way. It's annoying that these two don't live in the same package.